test_conformance/images/kernel_read_write/test_common.cpp

```diff
@@ -105,52 +130,94 @@ void InitFloatCoordsCommon(image_descriptor *imageInfo,
         i = 0;
         if (lod == 0)
         {
-            for (size_t z = 0; z < imageInfo->depth; z++)
+            for (size_t z = 0; z < depth_loop; z++)
             {
-                for (size_t y = 0; y < imageInfo->height; y++)
+                for (size_t y = 0; y < height_loop; y++)
                 {
-                    for (size_t x = 0; x < imageInfo->width; x++, i++)
+                    for (size_t x = 0; x < width_loop; x++, i++)
                     {
-                        xOffsets[i] = (float)((double)xOffsets[i]
-                                              / (double)imageInfo->width);
-                        yOffsets[i] = (float)((double)yOffsets[i]
-                                              / (double)imageInfo->height);
-                        zOffsets[i] = (float)((double)zOffsets[i]
-                                              / (double)imageInfo->depth);
+                        xOffsets[i] =
+                            (float)((double)xOffsets[i] / (double)width_loop);
+                        if (imageInfo->type != CL_MEM_OBJECT_IMAGE1D_ARRAY)
+                        {
+                            yOffsets[i] = (float)((double)yOffsets[i]
+                                                  / (double)height_loop);
+                        }
+                        if (imageInfo->type != CL_MEM_OBJECT_IMAGE2D_ARRAY)
+                        {
+                            zOffsets[i] = (float)((double)zOffsets[i]
+                                                  / (double)depth_loop);
+                        }
                     }
                 }
             }
         }
         else if (gTestMipmaps)
         {
-            size_t width_lod, height_lod, depth_lod;
-
-            width_lod =
-                (imageInfo->width >> lod) ? (imageInfo->width >> lod) : 1;
-            height_lod =
-                (imageInfo->height >> lod) ? (imageInfo->height >> lod) : 1;
-            depth_lod =
-                (imageInfo->depth >> lod) ? (imageInfo->depth >> lod) : 1;
+            width_loop = (width_loop >> lod) ? (width_loop >> lod) : 1;
```

Comment:
`lod` means Level Of Detail in that context. At first glance it seems like a meaninful name that should be kept.

Reply:
Added a commit that restores the _lod